feat: shift from cargo test to cargo nextest

Arya-A-Nair opened the pull request:
Shifting from cargo test to cargo nextest as per Issue- [#999](https://github.com/xmtp/libxmtp/issues/999)

insipx commented:
Hello!

Thanks for your external contribution!

I approved the workflows, and it seems the github workflow files need a bit more work:

```
error: no such command: `nextest`
	Did you mean `test`?
	View all installed commands with `cargo --list`
	Find a package to install `nextest` with `cargo search cargo-nextest`
```

insipx reviewed (changes requested):
CI

mkysel commented on Dockerfile:
```diff
@@ -8,6 +8,7 @@ COPY --chown=xmtp:xmtp rust-toolchain .
 ENV RUSTUP_PERMIT_COPY_RENAME "yes"
 
 RUN rustup update
+RUN curl -LsSf https://get.nexte.st/latest/linux | tar zxf - -C ${CARGO_HOME:-~/.cargo}/bin
```
@insipx are we using the dev container?

insipx replied:
I don't use it myself, but unsure if anyone else does

Arya-A-Nair replied:
The readme file stated to use the vscode dev container, even when I had to run the nextest command locally I had to install it, so for easier developer access I included it in the docker file

Arya-A-Nair commented:
> Hello!
> 
> Thanks for your contribution!
> 
> I approved the workflows, and it seems the github workflow files need a bit [more work](https://nexte.st/docs/installation/pre-built-binaries/#using-nextest-in-github-actions):
> 
> ```
> error: no such command: `nextest`
> 	Did you mean `test`?
> 	View all installed commands with `cargo --list`
> 	Find a package to install `nextest` with `cargo search cargo-nextest`
> ```

I had added the installation for cargo nextest in the docker file, but forgot to put the same in workflow, I have corrected the mistake in the latest commit

mkysel commented:
I would use https://github.com/taiki-e/install-action to install the tooling. The nextest examples themselves use it https://github.com/nextest-rs/reuse-build-partition-example/blob/main/.github%2Fworkflows%2Fci.yml

Arya-A-Nair commented:
> I would use https://github.com/taiki-e/install-action to install the tooling. The nextest examples themselves use it https://github.com/nextest-rs/reuse-build-partition-example/blob/main/.github%2Fworkflows%2Fci.yml

Hey @mkysel , I have replaced that particular piece of code in the latest commit

mkysel commented:
Thanks @Arya-A-Nair. Looks good to me. I will wait for @insipx to approve since he is the original reporter.

insipx reviewed (approved):
Looks great! Thank you

Arya-A-Nair commented:
> Looks great, thanks again for the contribution!
> 
> feel free to merge whenever

Hey, as I don't have write access I cant merge from my end

mkysel commented:
Done!
